music21/roman.py

```diff
@@ -2334,10 +2346,9 @@ def _setImpliedQualityFromString(self, workingFigure):
             workingFigure = workingFigure[1:]
             impliedQuality = 'augmented'
             # impliedQualitySymbol = '+'
-        elif workingFigure.endswith('d7'):
+        elif m := re.match(r'(?P<leading>.*)d(?P<figure>7|6/?5|4/?3|4/?2|2)$', workingFigure):
```

Comment:
rewrite as:

`elif 'd' in workingFigure and (m := ...what_you_currently_have...):`

because regular expression matching is slow, so we only want to do a regular expression match if there is already a 'd' there.  It will make it slightly slower for d65 etc. chords, but much faster for all other chords.

Reply:
In fact, it'd probably be better to throw exception on chords that can't be made dominant 7th chords:

```
if 'd' in workingFigure:
    m = re.match(r'(?P<leading>.*)d(?P<figure>7|6/?5|4/?3|4/?2|2)$', workingFigure)
    if m is None:
        raise RomanException(
              f'Cannot make a dominant-seventh chord out of {workingFigure}.'
        )
    ...rest of your code...
```

So, no, I don't want implied quality for things that aren't dominant sevenths.  If someone wants that they can use the "add" --  RomanNumeral('IVd65[add2]') or set the `implied_quality` later.